compliance_checker/cf/cf.py

```diff
@@ -3099,14 +3099,20 @@ def check_cf_role(self, ds):
         """
         ret_val = []
         valid_roles = ['timeseries_id', 'profile_id', 'trajectory_id']
+        variable_count = 0
         for variable in ds.get_variables_by_attributes(cf_role=lambda x: x is not None):
+            variable_count += 1
             name = variable.name
             valid_cf_role = TestCtx(BaseCheck.HIGH, '§9.5 {} contains a valid cf_role attribute'.format(name))
             cf_role = variable.cf_role
             valid_cf_role.assert_true(cf_role in valid_roles,
                                       "{} is not a valid cf_role value. It must be one of {}"
                                       "".format(name, ', '.join(valid_roles)))
-        return ret_val
+        valid_cf_role.assert_true(variable_count < 2,
+                                  ('§9.5 states that datasets should not '
+                                   'contain more than two variables defining a '
+                                   'cf_role attribute.'))
+        return valid_cf_role.to_result()
```

Comment:
The first iteration I wrote did that but what if the user incorrectly labeled the feature? I wrote this check from the perspective of, "this dataset is horribly formatted and we can only try to make certain guesses about how it _should_ be." If the checker always got perfectly formatted data, there would not be a need for a checker. 

I leave the rest of the feature detection stuff up to the individual feature checks.